scala/bigtable-shc/README.md

```diff
@@ -1,53 +1,71 @@
+
+**Apache Spark - Apache HBase Connector** allows the use of Spark SQL relational queries on data stored in Cloud Bigtable.
+
+The project contains the following sample applications:
+
+1. [BigtableSource.scala](src/main/scala/com/example/bigtable/spark/shc/BigtableSource.scala) shows how to write and execute
+SQL queries against data defined in a Bigtable database.
+
+1. [AgeSource.scala](src/main/scala/com/example/bigtable/spark/shc/AgeSource.scala) also shows how to write and execute SQL queries against data defined in a Bigtable database with other example usages that may be easier to understand.
 
 ## Prerequisites
 
 1. A [Google Cloud project](https://console.cloud.google.com/) with billing enabled. Please
 be aware of [Bigtable](https://cloud.google.com/bigtable/pricing)
 and [Dataproc](https://cloud.google.com/dataproc/docs/resources/pricing) pricing.
 
-1. You have the [Google Cloud SDK](https://cloud.google.com/sdk/) installed.
+1. [Google Cloud SDK](https://cloud.google.com/sdk/) installed.
 
-1. (Optional but recommended) You have [Scala](https://www.scala-lang.org/) installed.
+1. [Apache Maven](https://maven.apache.org/) installed.
 
-1. You have basic familiarity with Spark and Scala. It may be helpful to
-install Spark locally.
+1. [Apache Spark](https://spark.apache.org/) installed.
 
-1. You have [Maven](https://maven.apache.org/) installed.
+1. A basic familiarity with [Apache Spark](https://spark.apache.org/) and [Scala](https://www.scala-lang.org/).
 
-## Configure the CLI
+## Configuring CLI and Authentication
 
-Make sure you are using the correct Google Cloud Project:
+Make sure you are using the correct Google Cloud project:
 
     gcloud config set project your-project-id
 
-Either use the `application-default login` command or use a Service Account. See
-the docs on [authentication](https://cloud.google.com/docs/authentication/).
+Authenticate to a Google Cloud Platform API using service or user accounts.
 
-## Create a Cloud Bigtable Instance
+**NOTE**: In most situations, we recommend using a service account for authentication to a Google Cloud Platform (GCP) API.
+
+**NOTE**: When using end user credentials to access resources within your project, you must grant the user access to resources within your project. Do this in GCP by setting a role in Google Cloud Identity and Access Management (Cloud IAM).
+
+Learn about [authenticating to a GCP API](https://cloud.google.com/docs/authentication/) in the Google Cloud documentation.
+
+## Creating Cloud Bigtable Instance
 
 If you don't already have a Cloud Bigtable instance, create one
 
      gcloud beta bigtable instances create test-instance --cluster test-cluster --cluster-zone us-east1-b --cluster-num-nodes 3
 
-## Create a Cloud Dataproc Cluster
+## Creating Cloud Dataproc Cluster
 
 Create a Cloud Dataproc instance:
 
     gcloud dataproc clusters create spark-cluster
 
-## Running the sample
+## Running Example
```

Comment:
"Run the sample"

Reply:
I don't think this section has any info about running the sample. It just seems like repeated info from the overview. Perhaps this section could just be merged into that.